src/crypto.ts

```diff
@@ -226,37 +228,49 @@ export default class Crypto {
       }
     })
 
-    const downloadPromises : Array<Promise<void>> = []
-    for (let fieldId in attachmentRecords) {
-      // Original name for the file is not found
-      if (filenames[fieldId] === undefined) return null
-
-      downloadPromises.push(
-        axios.get(attachmentRecords[fieldId], { responseType: 'json' })
-      .then((downloadResponse) => {
-        const encryptedAttachment: EncryptedAttachmentContent = downloadResponse.data
-        const encryptedFile: EncryptedFileContent = {
-          submissionPublicKey: encryptedAttachment.encryptedFile.submissionPublicKey,
-          nonce: encryptedAttachment.encryptedFile.nonce,
-          binary: decodeBase64(encryptedAttachment.encryptedFile.binary),
-        }
-
-        return this.decryptFile(formSecretKey, encryptedFile)
-      }).then((decryptedFile) => {
-        if (decryptedFile === null) throw new Error("Attachment decryption failed")
-        decryptedRecords[fieldId] = { filename: filenames[fieldId], content: decryptedFile }
-      }))
+    const fieldIds = Object.keys(attachmentRecords)
+    // Check if all fieldIds are within filenames
+    if (!areAttachmentFieldIdsValid(fieldIds, filenames)) {
```

Comment:
this seems important - maybe we could add unit tests to make sure it is called + behaves correctly?

Reply:
there's an existing test for this: https://github.com/opengovsg/formsg-javascript-sdk/blob/8dd992272e1520b6d09b7d1b23b932863fde5fd7/spec/crypto.spec.ts#L350

Reply:
could i clarify if this means that the helper should be unit tested? there's an existing test which @mantariksh linked which checks this case

Reply:
as discussed, to add a small unit test for `areAttachmentFieldIdsValid`